### PR TITLE
JobAbsrtaction refactored to separate files

### PR DESCRIPTION
**Pull request by DunderRoffe**
Moved the classes JobAbstraction, CommandJob and FunctionJob from libsoftwarecontainer.h
to spearate .h files and moved the definitions from the header files into cpp files.

### CODE REVIEW

File: libsoftwarecontainer/include/commandjob.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+#ifndef SOFTWARECONTAINER_COMMANDJOB
```
Comment (breakreturn) [inline libsoftwarecontainer/include/commandjob.h]: We're mixing "#pragma once" and this style. Perhaps pragma once should be preferred?

Comment (tobsan) [reply libsoftwarecontainer/include/commandjob.h]: +1

**Comment by tobsan** [comment]
looks good

File: libsoftwarecontainer/include/commandjob.h (hunk 1 above)
Comment (DunderRoffe) [reply libsoftwarecontainer/include/commandjob.h]: should be fixed in the latest commit

**Pull request merged**